src/components/pages/Header.jsx

```diff
@@ -0,0 +1,97 @@
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+import closeIcon from '../../assets/icons/close.png';
+import loginIcon from '../../assets/icons/login.png';
+import logoutIcon from '../../assets/icons/logout.png';
+import navIcon from '../../assets/icons/nav.png';
+import searchIcon from '../../assets/icons/search.png';
+
+const Header = () => {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const handleToggle = () => {
+        setIsOpen(!isOpen);
+    };
+    
+    const location = useLocation();
```

Comment:
Yes, it is.

Reply:
There are a few things we have to consider. First off, there is a lot going on here so there is potential to refactor certain parts into different functions or even components. Consider this, if something is intended to be rendered or used in multiple components that alone can be it's own modular component/function.  That way all you need to do is plug it where you want instead of writing out all the code again. 

Second, you are directly manipulating the DOM. You do not want to do this. React has a special set of tools that you need to take advantage of when you want to dynamically adjust visibility. E.g. something like the "login button" you would only want to render if a user is not logged in. Where could you find this data? And how can you use that to conditionally render something?

note: currently there is no user data so we just need some generic placeholder that dictates if something should be rendered or not. We can incorporate that logic later on.